crates/client/rpc/src/versions/v0_8_0/methods/ws/lib.rs

```
@@ -0,0 +1,398 @@
+use std::borrow::Cow;
+
+use crate::{
+    errors::{StarknetWsApiError, WsResult},
+    versions::v0_8_0::StarknetWsRpcApiV0_8_0Server,
+};
+
+use super::BLOCK_PAST_LIMIT;
+
+#[jsonrpsee::core::async_trait]
+impl StarknetWsRpcApiV0_8_0Server for crate::Starknet {
+    async fn subscribe_new_heads(
```

everywhere here in this func you're dropping the Err values instead of displaying them as an internal server error

id advise you to either
```rs
.map_err(|e| {
  display_internal_server_error(format!("some context blah blah: {err:#}");
  StarknetWsApiError::InternalServerError
})
```
for Results, and something similar for unwrapping Options where None means an error.

A better way to do this is to do like we already do in the RPCs: `or_internal_server_error` / `or_else_internal_server_error` for unwrapping results, and `ok_or_internal_server_error` / `ok_or_else_internal_server_error` for unwrapping options.
You can search the codebase to see how they work, but that's the best way to handle internal servers errors in RPCs IMO.

Note: we don't want to show the reason of an internal server error to the client! as a general rule, it's not a good idea - only to the console

---

```rs
let Ok(sink) = pending.accept().await else {
  return WsResult::Err(StarknetWsApiError::internal(Cow::from("Failed to establish websocket connection")));
```
would look like
```rs
let sink = pending.accept().await
  .or_internal_server_error("Failed to establish websocket connection")?;
```

and 
```rs
let Ok(Some(block_info)) = self.backend.get_block_info_from_block_latest() else {
    return WsResult::Err(StarknetWsApiError::internal(Cow::from(format!(
        "Failed to retrieve block info for block {block_n}",
    ))));
};
```
would look like
```rs
let block_info = self.backend.get_block_info(xxx)
  .or_else_internal_server_error(|| format!("Failed to retrieve block info for block {block_n}"))?
  .ok_or_internal_server_error("There are no blocks in the database")?;
```
which, now that i realise, is not an internal server error at all - you should return a clean error if there is no latest block, which can only mean there are no blocks in db yet. For the standard rpc api there's a NoBlocks error for that.

---

This is already handled [here](https://github.com/madara-alliance/madara/blob/0b0cf309e31b7cde6f533b9d5c693bd25f02e106/crates/client/rpc/src/errors.rs#L237) but I think I'll more this code to the init of `InternalError` instead of the `message` function.